src/contracts/PerformanceFee.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {IERC20} from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import {MultiLP} from "./MultiLP.sol";
+
+/**
+ * @title Added a performance fee to an ARM with Liquidity Providers (LP)
+ * @author Origin Protocol Inc
+ */
+abstract contract PerformanceFee is MultiLP {
+    /// @notice The maximum performance fee that can be set
+    /// 10,000 = 100% performance fee
+    uint256 public constant MAX_FEE = 10000;
+
+    /// @notice The account that can collect the performance fee
+    address public feeCollector;
+    /// @notice Performance fee that is collected by the feeCollector measured in basis points (1/100th of a percent)
+    /// 10,000 = 100% performance fee
+    /// 2,000 = 20% performance fee
+    /// 500 = 5% performance fee
+    uint16 public fee;
+    /// @notice The performance fees accrued but not collected.
+    /// This is removed from the total assets.
+    uint112 public feesAccrued;
+    /// @notice The total assets at the last time performance fees were calculated.
+    /// This can only go up so is a high watermark.
+    uint128 public lastTotalAssets;
```

Comment:
We are going to need _gaps in these inherited contracts.

Reply:
definitely. I'll add.

FYI, The existing AMM v1 only uses three storage slots outside the keccak slots. 
https://explorer.sim.io/eth/20766860/0x85b78aca6deae198fbf201c82daf6ca21942acc6#map

I think we can just repurpose the slots in the new contract, but let's double check once we narrow down on what implementation we want to use for Lido